File: lab6_5.py
list1=[1,2,3,4]
list2=[3,4,5,6]
def function1(list1,list2):
    list3 = []
    list1.extend(list2)
    for x in list1:
        if(list1.count(x)>1 and list3.__contains__(x)==False):
            list3.append(x)
    return list3

print(function1(list1,list2))

list=["mom","nun","car","pencil","madam","door"]
def function2(list):
    plist=[]
    for x in list:
        if(x==x[::-1]):
            plist.append(x)
    return plist

print(function2(list))

primelist=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]
def function3(primelist):
    newlist=[]
    length=len(primelist)
    prime = [True for i in range(length + 1)]
    p = 2
    while (p * p <= length):
        if (prime[p] == True):
            for i in range(p * p, length + 1, p):
                prime[i] = False
        p += 1
    for p in range(2, length + 1):
        if prime[p]:
            newlist.append(p)
    return newlist

print(function3(primelist))

word="cinema"
word_list=["car","iceman","pencil","door","nemaci","macine"]
def anagrams(word,word_list):
    wList=[]
    wList2 = []
    outputList=[]
    for x in word:
        wList.append(x)
    wList.sort()
    for x in word_list:
        for y in x:
            wList2.append(y)
        wList2.sort()
        if(wList==wList2):
            outputList.append(x)
        wList2=[]
    return outputList

print(anagrams(word,word_list))

File: lab6_5.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;


vector<int> function1(vector<int> list1,vector<int> list2) {
    vector<int> list3;
    set_intersection(list1.begin(),list1.end(),list2.begin(),list2.end(),back_inserter(list3));
    return list3;
}
bool checkPalindrome(string palindrome) {
    int length=palindrome.length();
    for (int i=0;i<length/2;i++) {
        if (palindrome[i]!=palindrome[length-i-1]) {
            return false;
        }
    }
    return true;
}

vector<string> function2(vector<string> plist) {
    vector<string> newList;
    for (string i:plist) {
        if (checkPalindrome(i)) {
            newList.push_back(i);
        }
    }
    return newList;
}

bool checkPrime(int prime) {
    if (prime<2) {
    return false;
    }
    int length=(sqrt(prime));
    for (int i=2;i<=length;i++) {
        if (prime%i==0) {
        return false;
        }
    }
    return true;
}

vector<int> function3(vector<int> primeList) {
    vector<int> prime;
    for (int i=0;i<primeList.size();i++) {
        if (checkPrime(primeList[i])) {
            prime.push_back(primeList[i]);
        }
    }
    return prime;
}

vector<string> anagrams(string word,vector<string> word_list) {
    vector<string> anagram;
    sort(word.begin(),word.end());
    for (string i:word_list) {
        string tmp=i;
            sort(i.begin(),i.end());
        if (word==i) {
            anagram.push_back(tmp);
        }
    }
    return anagram;
}



int main() {
    vector<int> list1={1,2,3,4};
    vector<int> list2={3,4,5,6};
    vector<int> funcOutput1=function1(list1,list2);
    for (int i=0;i<funcOutput1.size();i++) {
        cout <<funcOutput1[i]<< " ";
    }

    cout <<" "<<endl;
    vector<string> pList={"mom","nun","car","pencil","madam","door"};
    vector<string> funcOutput2=function2(pList);
    for (string i:funcOutput2) {
        cout<<i<<" ";
    }

    cout<<" "<<endl;
    vector<int> primeList = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20};
    vector<int> funcOutput3=function3(primeList);
    for (int i=0;i<funcOutput3.size();i++) {
        cout<<funcOutput3[i]<<" ";
    }

    cout<<" "<<endl;
    string word="cinema";
    vector<string> word_list={"car","iceman","pencil","door","nemaci","macine"};
    vector<string> anagramoutput=anagrams(word,word_list);
    for (string i:anagramoutput) {
        cout<<i<<" ";
    }

}


